04_numeronym/runnerdave/main_test.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("[a11y K8s abc]\n")
+	actual := strconv.Quote(buf.String())
+	t.Logf("expected:%s", expected)
+	t.Logf("actual:%s", actual)
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()")
+	}
+}
+
+func TestNumeronyms(t *testing.T) {
+	tables := []struct {
+		x []string
+		n []string
+	}{
+		{[]string{"accessibility", "Kubernetes", "abc"}, []string{"a11y", "K8s", "abc"}},
+		{[]string{"tres"}, []string{"t2s"}},
+		{[]string{"cuatrO"}, []string{"c4O"}},
+		{[]string{}, []string{}},
+		{[]string{"11111111"}, []string{"161"}},
+		{[]string{"?"}, []string{"?"}},
+		{[]string{"?$"}, []string{"?$"}},
+		{[]string{"?$&%"}, []string{"?2%"}},
+	}
+
+	for _, table := range tables {
+		numys := numeronyms(table.x...)
+		if !equal(numys, table.n) {
+			t.Errorf("Sorting of (%v) was incorrect, got: %v, want: %v.", table.x, numys, table.n)
```

Comment:
I think this error message might not be indicative of the true failure. What if it were the case that the test failed because of the length's not being the same? Then it would still say "Sorting of ... was incorrect.... " I think it is fine to remove the "sorting" and just say "Unexpected output, expected ..., got ..."

Reply:
You are right, fixed